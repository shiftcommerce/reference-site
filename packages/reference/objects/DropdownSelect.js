// Libraries
import { Component } from 'react'

class DropdownSelect extends Component {
  constructor (props) {
    super(props)
    this.renderOptions = this.renderOptions.bind(this)
  }

  renderOptions (options) {
    return (
      options && options.map((option, idx) =>
        <option role='option' key={idx} value={option} aria-setsize={options.length} aria-posinset={idx + 1} >{ option }</option>
      )
    )
  }

  render () {
    let {
      name,
      prompt,
      options,
      ...otherProps
    } = this.props

    return (
      <select name={name} {...otherProps} role='listbox'>

        <option role='option' value='' aria-setsize={options.length} aria-posinset='0'>{ prompt }</option>

        { this.renderOptions(options) }

      </select>
    )
  }
}

export default DropdownSelect
