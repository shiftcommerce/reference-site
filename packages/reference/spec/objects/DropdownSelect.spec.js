// Object
import DropdownSelect from '../../objects/DropdownSelect'

test('renders correctly', () => {
  // Arrange
  const quantityOptions = [ 1, 2, 3, 4, 5 ]

  // Act
  const wrapper = mount(
    <DropdownSelect name='line_item[unit_quantity]' prompt='Select a Quantity' options={quantityOptions} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
})
