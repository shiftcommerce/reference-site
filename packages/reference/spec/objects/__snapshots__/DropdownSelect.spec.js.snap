// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DropdownSelect
  name="line_item[unit_quantity]"
  options={
    Array [
      1,
      2,
      3,
      4,
      5,
    ]
  }
  prompt="Select a Quantity"
>
  <select
    name="line_item[unit_quantity]"
    role="listbox"
  >
    <option
      aria-posinset="0"
      aria-setsize={5}
      role="option"
      value=""
    >
      Select a Quantity
    </option>
    <option
      aria-posinset={1}
      aria-setsize={5}
      role="option"
      value={1}
    >
      1
    </option>
    <option
      aria-posinset={2}
      aria-setsize={5}
      role="option"
      value={2}
    >
      2
    </option>
    <option
      aria-posinset={3}
      aria-setsize={5}
      role="option"
      value={3}
    >
      3
    </option>
    <option
      aria-posinset={4}
      aria-setsize={5}
      role="option"
      value={4}
    >
      4
    </option>
    <option
      aria-posinset={5}
      aria-setsize={5}
      role="option"
      value={5}
    >
      5
    </option>
  </select>
</DropdownSelect>
`;
